Add Foreman and Overview docs (#1420)
theforeman/foreman-infra: docs/overview.md

# Overview

> The Foreman project runs a number of different servers for testing, packaging, and continuous integration. Most of these servers are available as Jenkins build slaves for http://ci.theforeman.org.

Reviewer: ```suggestion
The Foreman project runs a number of different servers for testing, packaging, and continuous integration. Most of these servers are available as Jenkins build nodes for https://ci.theforeman.org.
```

## Access

Access to Foreman project infrastructure is available for those who wish to assist in building packages, testing, and building Jenkins jobs. Fork http://github.com/theforeman/foreman-infra and add an ssh_user resource to the users module (see puppet/modules/users/manifests/init.pp) and your key into the files directory. Submit a pull request to the infrastructure project and then talk to ? in #theforeman-dev on irc.freenode.net. One of them can merge your change and update the puppetmaster.

## Landscape

| Host | Provider | Function |
|---|---|---|
| ci.theforeman.org | OSUOSL | Jenkins |
|   |   |   |
|   |   |   |

## Infrastructure providers

A list of the hosting we have, who provides it, and what capacity it has

* Rackspace

> ** $2k per month

Reviewer: No longer sponsored so $0?

> ** Hosts web02, the Elastic logging instance, Jenkins, and some permanent slaves

Reviewer: I think the logging instance and slaves were removed, but the Foreman instance is there as well

** Contact support from our account as needed
* Scaleway

> ** $25 per month (not a lot!), but they're cheap so we do OK. Hosts Discourse, Redmine, and some ARM build slaves

Reviewer: I think this sponsoring has gone away so $0?

** Support usually helpful, Edouard Bonlieu & Yann Léger were initial contacts
* Fastly

> ** $1k/month CDN

Reviewer: We should mention that this is sponsored

** Elaine Greenberg was initial contact
* OSUOSL

> ** Free

Reviewer: ```suggestion
** Sponsored
```

> ** Hosts test machines, slaves, and stats.tf.o

Reviewer: ```suggestion
** Hosts test machines, Jenkins nodes and web01
```

** Contact Lance Alberston if more capacity is needed
* NETWAYS

> ** Free

Reviewer: Sponsored

> ** Contributing a single VM for the Builders/RH hostgroup

Reviewer: We actually have an account in their openstack instance with capacity for more than one. I just had issues installing and haven't gotten back to their support.

> ** Option to add an Icinga Monitoring host here, talk to Dirk to progress

Reviewer: Currently blocked on me responding to a support ticket

# Overview

The Foreman project runs a number of different servers for testing, packaging, and continuous integration.

## Access

> Access to Foreman project infrastructure is available for those who wish to assist in building packages, testing, and building Jenkins jobs. Fork http://github.com/theforeman/foreman-infra and add an ssh_user resource to the users module (see puppet/modules/users/manifests/init.pp) and your key into the files directory. Submit a pull request to the infrastructure project and then talk to ? in #theforeman-dev on irc.freenode.net. One of them can merge your change and update the puppetmaster.

Reviewer 2: ```suggestion
Access to Foreman project infrastructure is available for those who wish to assist in building packages, testing, and building Jenkins jobs.

Fork https://github.com/theforeman/foreman-infra and add your key into the [files directory](https://github.com/theforeman/foreman-infra/tree/master/puppet/modules/users/files) of the [`users` module](https://github.com/theforeman/foreman-infra/blob/master/puppet/modules/users/). Submit a pull request to the infrastructure project and then talk to ? in #theforeman-dev on irc.freenode.net. One of them can merge your change and add your user in the Foreman web UI.
```

Reviewer: Perhaps a Discourse post is a better suggestion?

Reviewer 2: Also that, yeah.

## Landscape

| Role | Provider |
|---|---|
| [Jenkins](./docs/jenkins.md) | OSUOSL |

> | [Koji](./docs/koji.md) | AWS |

Reviewer: While these have a single provider, Jenkins nodes won't.

|   |   |

## Infrastructure providers

A list of the hosting we have, who provides it, and what capacity it has

> * Rackspace
> ** 0k/month
> ** Hosts web02, Jenkins
> ** Contact support from our account as needed

Reviewer 2: ```suggestion
* Rackspace
  * 0k/month
  * Hosts web02, Jenkins, Foreman/Puppet
  * Contact support from our account as needed
```
the formating applies to the whole section  ;-)

* Scaleway
** 0k/month

> ** Support usually helpful, Edouard Bonlieu & Yann Léger were initial contacts

Reviewer 2: ```suggestion
  * Support usually helpful, Edouard Bonlieu & Yann Léger were initial contacts
  * Hosts Discourse, Redmine, ARM Jenkins nodes (EOL)
```

* Fastly
** Sponsored
** $1k/month CDN
** Elaine Greenberg was initial contact
* OSUOSL
** Sponsored
** Hosts test machines, Jenkins nodes, and web01
** Contact Lance Alberston if more capacity is needed
* NETWAYS
** Sponsored
** Openstack instance for spinning up as needed compute
** Option to add an Icinga Monitoring host here, talk to Dirk to progress
* Gandi

> ** Provides DNS

Reviewer 2: sponsors theforeman.org domain

## Access

Access to Foreman project infrastructure is available for those who wish to assist in building packages, testing, and building Jenkins jobs.

Fork https://github.com/theforeman/foreman-infra and add your key into the [files directory](https://github.com/theforeman/foreman-infra/tree/master/puppet/modules/users/files) of the [`users` module](https://github.com/theforeman/foreman-infra/blob/master/puppet/modules/users/). Submit a pull request to the infrastructure project and send a post to the [Infra SIG discourse thread](https://community.theforeman.org/c/development/infra/24). One of them can merge your change and add your user in the Foreman web UI.

## Landscape

| Role | Provider(s) |
|---|---|
| [Jenkins](./docs/jenkins.md) | OSUOSL |
| [Koji](./docs/koji.md) | AWS |
| [Jenkins Nodes](./docs/jenkins.md)  | OSUOSL / AWS |

## Infrastructure providers

A list of the hosting we have, who provides it, and what capacity it has

* Rackspace
  * Previously sponsored, [being phased out](https://community.theforeman.org/t/rfc-moving-off-of-rackspace-infrastructure/17932)

>   * Hosts web02, Jenkins

Reviewer: web02 was turned off

Access to Foreman project infrastructure is available for those who wish to assist in building packages, testing, and building Jenkins jobs.

Fork https://github.com/theforeman/foreman-infra and add your key into the [files directory](https://github.com/theforeman/foreman-infra/tree/master/puppet/modules/users/files) of the [`users` module](https://github.com/theforeman/foreman-infra/blob/master/puppet/modules/users/). Submit a pull request to the infrastructure project and send a post to the [Infra SIG discourse thread](https://community.theforeman.org/c/development/infra/24). One of them can merge your change and add your user in the Foreman web UI.

## Landscape

| Role | Provider(s) |
|---|---|
| [Jenkins](./docs/jenkins.md) | OSUOSL |
| [Koji](./docs/koji.md) | AWS |
| [Jenkins Nodes](./docs/jenkins.md)  | OSUOSL / AWS |

## Infrastructure providers

A list of the hosting we have, who provides it, and what capacity it has

* Rackspace

>   * 0k/month

Reviewer: This is a bit odd
```suggestion
  * Previously sponsored, [being phased out](https://community.theforeman.org/t/rfc-moving-off-of-rackspace-infrastructure/17932)
```

  * Hosts web02, Jenkins
  * Contact support from our account as needed
* Scaleway

>   * 0k/month

Reviewer: ```suggestion
  * Previously sponsored, planned to be phased out
```

  * Support usually helpful, Edouard Bonlieu & Yann Léger were initial contacts

>     * Hosts Discourse, Redmine, ARM Jenkins nodes

Reviewer 2: ```suggestion
  * Hosts Discourse, Redmine, ARM Jenkins nodes
```

# Overview

The Foreman project runs a number of different servers for testing, packaging, and continuous integration. Most of these servers are available as Jenkins build slaves for http://ci.theforeman.org.

## Access

Access to Foreman project infrastructure is available for those who wish to assist in building packages, testing, and building Jenkins jobs. Fork http://github.com/theforeman/foreman-infra and add an ssh_user resource to the users module (see puppet/modules/users/manifests/init.pp) and your key into the files directory. Submit a pull request to the infrastructure project and then talk to ? in #theforeman-dev on irc.freenode.net. One of them can merge your change and update the puppetmaster.

## Landscape

| Host | Provider | Function |
|---|---|---|
| ci.theforeman.org | OSUOSL | Jenkins |
|   |   |   |

> |   |   |   |

Reviewer: I think listing the individual hosts is not that important and what we have Foreman for. IMHO a list of roles (foreman, web, Jenkins master and Jenkins node) is more important. They can then link to the individual detailed pages.
